podcasts/Enumerations.swift

```diff
@@ -261,13 +266,27 @@ enum PlayerAction: Int, CaseIterable, AnalyticsDescribable {
             return "shelf_played"
         case .archive:
             return episode is UserEpisode ? "shelf_delete" : "shelf_archive"
+        case .addBookmark:
+            return "bookmarks-shelf-icon"
         }
     }
 
     func canBePerformedOn(episode: BaseEpisode) -> Bool {
         switch self {
         case .starEpisode, .shareEpisode:
             return episode is Episode
+        case .addBookmark:
+            return isAvailable
+        default:
+            return true
+        }
+    }
+
+    /// Return false to completely hide the action from the shelf and overflow menu
```

Comment:
[nit] This comment made me think for a few seconds. Maybe we can change it to something like "If false, then it will hide the action from the shelf and overflow menu"?

Reply:
Improved it here: 2746d0f8dc4ca7b7e50d562932bfc1c3e9dcb008